runtime/appruntime/api/handler.go

```diff
@@ -358,7 +376,10 @@ func (d *Desc[Req, Resp]) Call(c CallContext, req Req) (resp Resp, respErr error
 
 		if rpcErr == nil {
 			r, rpcErr = d.CloneResp(r)
-			httpStatus = errs.HTTPStatus(errs.Convert(rpcErr))
+			if rpcErr != nil {
+				// only override the http status if an error occurred trying to clone the response
+				httpStatus = errs.HTTPStatus(errs.Convert(rpcErr))
+			}
```

Comment:
Interestingly we always overrode the httpStatus after cloning the response, which means if middleware had set `201` or something, we'd reset it to `200`